src/tests/adr/strings_adr.rs

```diff
@@ -0,0 +1,132 @@
+use crate::test_utils::tests::codegen;
+
+/// # Architecture Design Record: Strings
+/// ST supports two types of Strings: UTF8 and UTF16 Strings. Strings are fixed size and
+/// stored using i8-arrays for utf8 strings and i16-arrays for utf16 strings.
+#[test]
+fn declaring_a_string() {
+    // two string variables
+    let src = r#"
+        VAR_GLOBAL
+            myUtf8  : STRING[20];
+            myUtf16 : WSTRING[20];
+        END_VAR
+        "#;
+
+    // ... are stored as i8/i16 arrays and get initialized to blank (0)
+    insta::assert_snapshot!(codegen(src), @r###"
+    ; ModuleID = 'main'
+    source_filename = "main"
+
+    @myUtf8 = global [21 x i8] zeroinitializer
+    @myUtf16 = global [21 x i16] zeroinitializer
+    "###);
+}
+
+/// rusty treats strings like C-strings (char arrays) so the interoperability
+/// with C-libraries is seamless.
+#[test]
+fn strings_are_terminated_with_0byte() {
+    // two strings with an initial value ...
+    let src = r#"
+        VAR_GLOBAL
+            myUtf8  : STRING[5]  := 'Hello';
+            myUtf16 : WSTRING[5] := "World";
+        END_VAR
+        "#;
+
+    // ... get stored as c-like char arrays with 0-terminators
+    // ... offer one extra entry (length 21 while only 20 were declared) for a terminator
+    insta::assert_snapshot!(codegen(src), @r###"
+    ; ModuleID = 'main'
+    source_filename = "main"
+
+    @myUtf8 = global [6 x i8] c"Hello\00"
+    @myUtf16 = global [6 x i16] [i16 87, i16 111, i16 114, i16 108, i16 100, i16 0]
+    "###);
+}
+
+/// Arrays are aggregate types. This means that passing them to functions and assigning them
```

Comment:
Probably say Strings instead of arrays here

Reply:
done